src/mame/nmk/nmk16.cpp

```diff
@@ -4899,19 +4902,38 @@ void nmk16_state::bjtwin(machine_config &config)
 	nmk112.set_rom1_tag("oki2");
 }
 
-// the 215 writes minimal data here, probably only enables the decryption logic, rather than supplying the decryption table
+// the 215 writes minimal data here. Data is, in fact, used for selecting one of the internal configurations hardwired inside NMK214 device to make the decryption logic:
 void tdragon_prot_state::mcu_port3_to_214_w(u8 data)
 {
-	// startup only
-	logerror("%s: mcu_port3_to_214_w (data %02x)\n", machine().describe_context(), data);
+	LOG("%s: mcu_port3_to_214_w (data %02x)\n", machine().describe_context(), data);
+
+	// Startup only. Bit 2 on Port 3 of MCU (NMK215) acts as strobe/clock (rising edge) for storing the byte previously set on the Port 7 of MCU (NMK215) into the internal registers of both NMK214 devices
+	if (m_init_clock_nmk214 == 0 && BIT(data, 2) == 1)
+	{
+		// Value is sent to both devices at the same time. Internally, each one evaluates if the value should be stored or not based on bit 3 of the incoming value matches to the operation mode configured on each device:
+		m_nmk214[0]->set_init_config(m_init_data_nmk214);
+		m_nmk214[1]->set_init_config(m_init_data_nmk214);
+		
+		// TEMP: force decode gfx after setting both nmk214 init config, just for testing purposes. Real devices perform the decoding on the fly for each byte/word fetch from GFX ROMs
+		if (m_nmk214[0]->is_device_initialized() && m_nmk214[1]->is_device_initialized())
+			decode_sabotenb();
+	}
+
+	m_init_clock_nmk214 = BIT(data, 2);
 }
```

Comment:
Do you get a problem if the user soft resets the system (pressing F3 by default)?  Will it decode the ROMs a second time?

Reply:
You are right. GFX data is decoded again after a soft reset. I included an additional flag to control that the decoding process is already done and avoiding doing it again.
Anyway, this is a side-effect of decoding the entire-region when devices are initialized. If the decoding could be hooked up for each GFX data fetch (byte/word), this won't happen.